internal/remote/tcp_conn.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package remote
+
+import (
+	"context"
+	"net"
+	"sync"
+)
+
+var (
+// TODO: Adding TCP Connections Pool, https://github.com/apache/rocketmq-client-go/issues/298
+//pool = &tcpConnPool{}
+)
+
+type tcpConnPool struct {
+	count int
+	pool  map[string]net.Conn
+	mutex sync.Mutex
+}
+
+func (pool *tcpConnPool) get(addr string) *tcpConnWrapper {
+	pool.mutex.Lock()
+	defer pool.mutex.Unlock()
+	return nil
+}
+
+func (pool *tcpConnPool) putBack(wrapper tcpConnWrapper) {
+	pool.mutex.Lock()
+	defer pool.mutex.Unlock()
+
+}
+
+type tcpConnWrapper struct {
+	net.Conn
+	closed bool
```

Comment:
never use bool directly, use  Bool in "go.uber.org/atomic"

Reply:
done